sys/vm/vm_mmap.c

```diff
@@ -234,6 +234,35 @@ vm_wxcheck(struct proc *p, char *call)
 	return (0);
 }
 
+static inline int
+vm_check_prot(vm_prot_t prot, const char *func, const char *protname)
+{
+	if ((prot & (PROT_READ_CAP | PROT_READ)) == PROT_READ_CAP) {
+		SYSERRCAUSE("%s: PROT_READ_CAP without PROT_READ in %s",
+		    func, protname);
+		return (ENOTSUP);
+	}
+	if ((prot & (PROT_WRITE_CAP | PROT_WRITE)) == PROT_WRITE_CAP) {
+		SYSERRCAUSE("%s: PROT_WRITE_CAP without PROT_WRITE in %s",
+		    func, protname);
+		return (ENOTSUP);
+	}
+	/*
```

Comment:
I think this detail might be worth calling out explicitly in the commit log.  Arguably if we want to go this route, then I would actually suggest that we don't even expose separate R/W bits via mmap().  That is, I would just have `PROT_CAP` and `PROT_NO_CAP` exposed to userspace.  Potentially we could collapse this down further in the VM to a single "VM_PROT_CAP" as well without separate R/W bits there.